### PR TITLE
add unit test for fields.go, result.go, rows.go.

### PR DESCRIPTION
**Pull request by dongzl**
**What this PR does**:

**Which issue(s) this PR fixes**:

For #72 .

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Review by jjeffcaii — APPROVED** [review]
LGTM

**Review by cjphaha — APPROVED** [review]

**Comment by cjphaha** [comment]
LGTM

File: pkg/mysql/fields_test.go [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+	assert.Equal(t, "t_order", field.TableName())
+}
+
+func TestDataBaseName(t *testing.T) {
+	field := createDefaultField()
+	assert.Equal(t, "db_arana", field.DataBaseName())
+}
+
+func TestTypeDatabaseName(t *testing.T) {
+	field := createField(mysql.FieldTypeBit, mysql.Collations[mysql.DefaultCollation])
```
Comment (AlexStocks) [inline pkg/mysql/fields_test.go:45]: test such units in a loop

File: pkg/mysql/fields_test.go [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+	assert.Equal(t, "VARCHAR", field.TypeDatabaseName())
+
+	field = createField(mysql.FieldTypeVarString, mysql.Collations[mysql.BinaryCollation])
+	assert.Equal(t, "VARBINARY", field.TypeDatabaseName())
+
+	field = createField(mysql.FieldTypeVarString, mysql.Collations[mysql.DefaultCollation])
+	assert.Equal(t, "VARCHAR", field.TypeDatabaseName())
+
+	field = createField(mysql.FieldTypeYear, mysql.Collations[mysql.DefaultCollation])
+	assert.Equal(t, "YEAR", field.TypeDatabaseName())
```
Comment (luky116) [inline pkg/mysql/fields_test.go:148]: The above codes have the same format and can be replaced by loop. For example:
```go
	var tests = []struct {
		field     mysql.FieldType
		collation string
		want      string
	}{
		{mysql.FieldTypeBit, mysql.DefaultCollation, "BIT"},
		{mysql.FieldTypeBLOB, mysql.DefaultCollation, "TEXT"},
		{mysql.FieldTypeBLOB, mysql.BinaryCollation, "BLOB"},
		// ...
	}
	for _, tt := range tests {
		field := createField(tt.field, mysql.Collations[tt.collation])
		assert.Equal(t, tt.want, field.TypeDatabaseName())
	}

```

File: pkg/mysql/fields_test.go (hunk 1 above)
Comment (dongzl) [reply pkg/mysql/fields_test.go]: > test such units in a loop

OK, I fix it.

File: pkg/mysql/result_test.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+	insertId, err := result.LastInsertId()
+	assert.Equal(t, uint64(2000), insertId)
+	assert.True(t, err == nil)
+}
+
+func TestRowsAffected(t *testing.T) {
+	result := createResult()
+	affectedRows, err := result.RowsAffected()
+	assert.Equal(t, uint64(10), affectedRows)
+	assert.True(t, err == nil)
```
[the remark this answers is not in the document]
Comment (dongzl) [reply pkg/mysql/result_test.go]: > you can use this to assert nil
> 
> > assert.Nil(t, err)

OK, Thanks.

File: pkg/mysql/rows_test.go [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+}
+
+func TestDecodeForRow(t *testing.T) {
+	row := &Row{
+		Content:   createContent(),
+		ResultSet: createResultSet(),
+	}
+	val, err := row.Decode()
+	//TODO row.Decode() is empty.
+	assert.True(t, val == nil)
```
[the remark this answers is not in the document]
Comment (dongzl) [reply pkg/mysql/rows_test.go]: > you can use this to assert nil
> 
> > assert.Nil(t, err)

OK, Thanks.

File: pkg/mysql/fields_test.go (hunk 2 above)
Comment (dongzl) [reply pkg/mysql/fields_test.go]: > The above codes have the same format and can be replaced by loop. For example:
> 
> ```go
> 	var tests = []struct {
> 		field     mysql.FieldType
> 		collation string
> 		want      string
> 	}{
> 		{mysql.FieldTypeBit, mysql.DefaultCollation, "BIT"},
> 		{mysql.FieldTypeBLOB, mysql.DefaultCollation, "TEXT"},
> 		{mysql.FieldTypeBLOB, mysql.BinaryCollation, "BLOB"},
> 		// ...
> 	}
> 	for _, tt := range tests {
> 		field := createField(tt.field, mysql.Collations[tt.collation])
> 		assert.Equal(t, tt.want, field.TypeDatabaseName())
> 	}
> ```

OK, Thanks.

**Review by AlexStocks — APPROVED** [review]

**Pull request merged**